Source/Snowman.cpp

```diff
@@ -273,3 +290,135 @@ void Snowman::randomTranslationSnowman(GLFWwindow* window, const bool& shift, bo
+					  positionCollider.z - collider.getCollidableDimensions().z <= origin.z);
+
+		if (collisionX && collisionZ) {
+			colliders = { collider }; // only keep the colliding object
+			break;
+		}
+	}
+
+	collideLeftToRight = true; // for some reason it breaks if i don't add these lines
+	collideBackwards   = true;
+	collideRightToLeft = true;
+	collideForward     = true;
+
+	//std::cout << collisionX << " " << collisionZ << std::endl;
+	return collisionX && collisionZ;
+}
+
+/**
+* If the snowman is colliding, check if it is colliding on the x axis
+* Disabled the key associated that is going in the direction of the collision
+*/
+void Snowman::CheckCollisionX(std::vector<CollidableModel> colliders, bool isColliding) {
+
+	for (auto& collider : colliders) {
+		glm::vec3 positionCollider = collider.getColliderPosition();
+
+		// keyA
+		if (origin.x <= positionCollider.x && // check if the snowman is colliding with the right side (between origin and half its width)
+			origin.x + this->getDimensions().x / 2 >= positionCollider.x - collider.getCollidableDimensions().x / 2 && isColliding) {
+			collideLeftToRight = false;
+			collideRightToLeft = true;
+		}
+		// keyD
+		else if (origin.x >= positionCollider.x && //check if the snowman is colliding on the left side
+				 origin.x - this->getDimensions().x / 2 <= positionCollider.x + collider.getCollidableDimensions().x / 2 && isColliding) {
+			collideLeftToRight = true;
+			collideRightToLeft = false;
+		}
+		else {
+			collideLeftToRight = true;
+			collideRightToLeft = true;
+		}
+	}
+}
+
+/**
+* If the snowman is colliding, check if it is colliding on the z axis
+* Disabled the key associated that is going in the direction of the collision
+*/
+void Snowman::CheckCollisionZ(std::vector<CollidableModel> colliders, bool isColliding) {
+
+	for (auto& collider : colliders) {
+		glm::vec3 positionCollider = collider.getColliderPosition();
+
+		// keyW
+		if (origin.z <= positionCollider.z  && // activates in the top half of the collider
+			origin.z + this->getDimensions().z / 2 >= positionCollider.z - collider.getCollidableDimensions().z / 2 && isColliding) {
+
+			collideForward = false;
+			collideBackwards = true;
+		}
+		// keyS
+		else if (origin.z >= positionCollider.z && // activates in the bottom half of the collider
+				 origin.z - this->getDimensions().z / 2 <= positionCollider.z + collider.getCollidableDimensions().z / 2 && isColliding) {
+			collideForward = true;
+			collideBackwards = false;
+		}
+		else {
+			collideForward = true;
+			collideBackwards = true;
+		}
+	}
+}
+
+/**
+* Get the dimensions of the snowman
+* For now, it is hardcoded to the intialization of the colliderBox because decompose doesn't work for some reason
+* @return colliderBox dimensions
+*/
+glm::vec3 Snowman::getDimensions() {
+	return { 4.0f, 2.0f, 4.0f };
+}
+
+/**
+* Adjusts the keys that collider with a side of the box depending on the snowman's rotation
+*/
+void Snowman::adjustKeysToRotation() {
```

Comment:
not sure if this is done right, but we will be leaving this to a bug ticket since we could not figure it out within a few days